### PR TITLE
Add OCLIF manifests

### PR DESCRIPTION
**Pull request by shauns**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/internal-cli-foundations/issues/478

### WHAT is this pull request doing?

1. Adds refresh-manifests command to generate OCLIF manifests for our plug-ins
2. Checks if running this on CI dirties the repo -- if so, that's a fail

### How to test your changes?

Run `pnpm refresh-manifests` / look at the CI log

### Measuring impact

After our next release, confirm that manifest JSON files are included in the plug-in packages.

### CODE REVIEW

File: .github/workflows/shopify-cli.yml [hunk 1]
```diff
@@ -130,6 +130,33 @@ jobs:
...
+        uses: ./.github/actions/setup-cli-deps
+        with:
+          node-version: 18.7.0
+      - name: Build
+        run: pnpm build
+      - name: Refresh manifests
+        run: pnpm refresh-manifests
+      - name: Check if there are changes
+        id: changes
+        uses: UnicornGlobal/has-changes-action@v1.0.12
```
Comment (pepicrft) [inline .github/workflows/shopify-cli.yml:154]: What about scoping this check to the manifests? Some commands like `yarn install` against the integration app might lead to file changes that are unrelated to the manifests.

File: .github/workflows/shopify-cli.yml [hunk 2]
```diff
@@ -130,6 +130,33 @@ jobs:
       - name: ${{ matrix.target }}
         run: pnpm nx run-many --all --skip-nx-cache --target=${{ matrix.target }}
 
+  oclif-manifests:
+    name: Check OCLIF manifests
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request' }}
```
Comment (pepicrft) [inline .github/workflows/shopify-cli.yml:136]: Don't we want to run this job also in `main`?

**Review by pepicrft — APPROVED** [review]

**Review by isaacroldan — APPROVED** [review]
